src/Controller/Async/Stack.php

```diff
@@ -38,21 +41,28 @@ public function add(Request $request)
         $filename = $request->request->get('filename');
 
         $stack = $this->app['stack'];
+        $dispatcher = $this->app['dispatcher'];
 
         /** @var FileInterface|null $removed */
         $file = $stack->add($filename, $removed);
 
         $panel = $this->render('@bolt/components/stack/panel-item.twig', ['file' => $file]);
+        $panelEvent = new GetResponseForControllerResultEvent($this->app['kernel'], $request, HttpKernelInterface::SUB_REQUEST, $panel);
+        $dispatcher->dispatch(KernelEvents::VIEW, $panelEvent);
+
         $list = $this->render('@bolt/components/stack/list-item.twig', ['file' => $file]);
+        $listEvent = new GetResponseForControllerResultEvent($this->app['kernel'], $request, HttpKernelInterface::SUB_REQUEST, $list);
+        $dispatcher->dispatch(KernelEvents::VIEW, $listEvent);
```

Comment:
Haha … this is one of those that I was desperate for you to wake up to go over … let me expand …

We need to send a JSON response object … that, um, contains fields with HTML … C.D.O. level 11 but still. So the initial version of the commit just used Twig directly.

However, when reviewing my change(s), it dawned on me … `TemplateView`'s goal in life is to allow people to respond to render events, and while to some the Stack is a small feature, to others the can't live without it … or me who'd like to be able to remove it from **my** account's rendering … i.e. it is one of those renders we want to have `KernelEvents::VIEW` dispatched for.

I'm going to dig this morning, and see how much work to add a sub request, but TBH for 3.3 due out in days, I'm more than a little cautious about adding routes to do this at this stage … 3.4 a different story.